main_test.go

```diff
@@ -162,3 +162,91 @@ func TestAddPath(t *testing.T) {
 	expected := []string{"existing", "new"}
 	assert.Equal(t, expected, addPath(paths, "new"))
 }
+
+func TestLinterFlags(t *testing.T) {
```

Comment:
This might be more personal preference, but I generally like to keep the tests named after the function they are testing and with the condition at the end of the name. So in this case I think the tests should be:

```go
TestSetupFlagsLinterFlag
TestSetupFlagsConfigWithLinterString
TestSetupFlagsConfigWithLinterMap
TestSetupFlagsConfigAndLinterFlag
```

I think that makes it easier to find the tests which cover a specific function.